fl4health/feature_alignment/string_columns_transformer.py

```diff
@@ -0,0 +1,40 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from fl4health.feature_alignment.constants import TextFeatureTransformer
+
+
+class StringColumnTransformer(BaseEstimator, TransformerMixin):
+    """
+    The purpose of this class is to enable the application of text feature transformers from sklearn
+    to multiple string columns, which is not supported in the first place.
+    """
+
+    def __init__(self, transformer: TextFeatureTransformer):
+        self.transformer = transformer
+
+    def fit(self, X: pd.DataFrame, y: pd.DataFrame = None) -> "StringColumnTransformer":
+        joined_X = X.apply(lambda x: " ".join(x), axis=1)
+        self.transformer.fit(joined_X)
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        joined_X = X.apply(lambda x: " ".join(x), axis=1)
+        return self.transformer.transform(joined_X)
+
+
+class TextColumnTransformer(BaseEstimator, TransformerMixin):
+    """
+    The purpose of this class is to enable the application of text feature transformers from sklearn
+    to a single-column pandas dataframe, which is not supported in the first place.
+    """
+
+    def __init__(self, transformer: TextFeatureTransformer):
+        self.transformer = transformer
+
+    def fit(self, X: pd.DataFrame, y: pd.DataFrame = None) -> "StringColumnTransformer":
+        self.transformer.fit(X[X.columns[0]])
```

Comment:
If X is meant to be a DF with a single column, we should probably assert that this is true in both the fit and transform functions.